registration/forms.py

```diff
@@ -247,25 +247,32 @@ def register_form_generator(conf):
                 'label': label,
             }
 
-            if field_type == 'string' and choices and is_editable:
+            if field_type == 'string':
                 d['required'] = is_required
                 d['initial'] = initial
-                d['help_text'] = help_text
-                d['choices'] = choices
-                d['widget'] = django.forms.RadioSelect
-                field_class = django.forms.ChoiceField
-            elif field_type == 'string' and field_name == 'email':
-                d['required'] = is_required
-                d['max_length'] = max_length
-                d['initial'] = initial
-                d['help_text'] = help_text
-                field_class = django.forms.EmailField
-            elif field_type == 'string':
-                d['required'] = is_required
-                d['max_length'] = max_length
-                d['initial'] = initial
-                d['help_text'] = help_text
-                field_class = django.forms.CharField
+                if choices and is_editable:
+                    d['help_text'] = help_text
+                    d['choices'] = choices
+                    d['widget'] = django.forms.RadioSelect
+                    field_class = django.forms.ChoiceField
+                elif field_name == 'email':
+                    d['max_length'] = max_length
+                    d['help_text'] = help_text
+                    field_class = django.forms.EmailField
+                elif field_name == 'license_id'\
+                        and 'license_id_formats' in conf:
+                    d['max_length'] = max_length
+                    license_id_formats = '<p class=\'hint-license-id-format\'>' \
+                                         'Valid Formats: ' \
+                                         + ', '.join(map(str, conf['license_id_formats'])) + '</p>'
+                    help_text = '<p>' + unicode(help_text) + '</p>'
```

Comment:
@mabuaisha this shouldn't use ``+``. It should use string templates ("{}".format)